src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -32,7 +32,7 @@
 from torch.nn import CrossEntropyLoss
 
 from ...activations import ACT2FN
-from ...cache_utils import Cache, DynamicCache, StaticCache
+from ...cache_utils import Cache, DynamicCache, SinkCache, StaticCache
```

Comment:
`SinkCache` cache-specific changes should be handled entirely in `SinkCache`, otherwise we will need to change ALL models (present and future) 😬 

1. `input_ids = input_ids[:, -max_cache_length:]` -- we can make `SinkCache` ignore values out of bounds, kinda like you already did :D 
2. `cache_position = cache_position[:input_length]` -- I'm assuming this is mostly for `_update_causal_mask` 🤔 Can't we repurpose the `using_static_cache` branch?

Reply:
1. Yes, I think that can be done and that would require rerotations :)
2. Yes, it's to get the correct mask. I will see if it can be done from within `_update_causal_mask`